vassal-app/src/main/java/VASSAL/build/module/map/MoveCameraButton.java

```diff
@@ -0,0 +1,324 @@
+ * License along with this library; if not, copies are available
+ * at http://www.opensource.org.
+ */
+package VASSAL.build.module.map;
+
+import VASSAL.build.AbstractToolbarItem;
+import VASSAL.build.AutoConfigurable;
+import VASSAL.build.BadDataReport;
+import VASSAL.build.Buildable;
+import VASSAL.build.module.Map;
+import VASSAL.build.module.documentation.HelpFile;
+import VASSAL.configure.Configurer;
+import VASSAL.configure.ConfigurerFactory;
+import VASSAL.configure.FormattedExpressionConfigurer;
+import VASSAL.configure.PropertyExpression;
+import VASSAL.configure.TranslatableStringEnum;
+import VASSAL.configure.VisibilityCondition;
+import VASSAL.counters.SendToLocation;
+import VASSAL.i18n.Resources;
+import VASSAL.script.expression.AuditTrail;
+import VASSAL.tools.ErrorDialog;
+import VASSAL.tools.FormattedString;
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.awt.Point;
+import java.util.List;
+
+/**
+ * Adds a button to a Maps toolbar that moves the view/camera to a specific point
+ */
+public class MoveCameraButton extends AbstractToolbarItem {
+  public static String MOVE_CAMERA_MODE = "moveCameraMode"; //NON-NLS
+  public static String BOARD_NAME = "boardName"; //NON-NLS
+  public static String X_POS = "xPos"; //NON-NLS;
+  public static String Y_POS = "yPos"; //NON-NLS;
+  public static String ZONE_NAME = "zoneName"; //NON-NLS;
+  public static String GRID_LOCATION = "gridLocation"; //NON-NLS;
+  public static String REGION_NAME = "regionName"; //NON-NLS
+  public static String PROPERTY_FILTER = "propertyFilter"; //NON-NLS
+  public static String X_OFFSET = "xOffset"; //NON-NLS
+  public static String Y_OFFSET = "yOffset"; //NON-NLS
+
+  protected String moveCameraMode = SendToLocation.DEST_LOCATION;
+  protected FormattedString board = new FormattedString("");
+  protected FormattedString x = new FormattedString("");
+  protected FormattedString y = new FormattedString("");
+  protected FormattedString gridLocation = new FormattedString("");
+  protected FormattedString zone = new FormattedString("");
+  protected FormattedString region = new FormattedString("");
+  protected FormattedString xOffset = new FormattedString("0");
+  protected FormattedString yOffset = new FormattedString("0");
+  protected PropertyExpression propertyFilter = new PropertyExpression("");
+
+  protected Map map;
+
+  public static class EmptyFormatConfig implements ConfigurerFactory {
+    @Override
+    public Configurer getConfigurer(AutoConfigurable c, String key, String name) {
+      return new FormattedExpressionConfigurer(key, name, new String[] { });
+    }
+  }
+
+  public MoveCameraButton() {
+    setNameKey("");
+
+    setLaunchButton(makeLaunchButton(
+      Resources.getString("Editor.MoveCameraButton.move_camera_tooltip"),
+      Resources.getString("Editor.MoveCameraButton.move_camera"),
+      "/images/recenter.gif", //NON-NLS
+      e -> {
+        moveCamera();
+      }
```

Comment:
One liners you can do like this: `e -> moveCamera()`

Reply:
Yeah I probably deleted some other crap